src/libfetchers/git.cc

```diff
@@ -212,9 +210,13 @@ std::pair<StorePath, Input> fetchFromWorkdir(ref<Store> store, Input & input, co
     if (fetchSettings.warnDirty)
         warn("Git tree '%s' is dirty", workdir);
 
-    auto gitOpts = Strings({ "-C", workdir, "--git-dir", gitDir, "ls-files", "-z" });
+    auto gitOpts = Strings({ "-C", workdir, "--git-dir", gitDir, "ls-files", "--cached", "-z" });
     if (submodules)
         gitOpts.emplace_back("--recurse-submodules");
+	else {
+        gitOpts.emplace_back("--others");
+        gitOpts.emplace_back("--exclude-standard");
```

Comment:
Would it be possible to have an option to not run the standard excludes? I would like to be able to have `.gitignore`d local files that still get applied in my nix config. I don't care about other users/programs on the computer reading them, they just don't make sense to be in the upstream repo.

(Also see #12291 for an alternative approach.)